File: web/js/compile.js
//
function uncompile(code)
{
    var s="";
    for(var i=0;i<code.length;i+=4){
        var c=0;
        c+=(code.charCodeAt(i)-97) << 12;
        c+=(code.charCodeAt(i+1)-97) << 8;
        c+=(code.charCodeAt(i+2)-97) << 4;
        c+=(code.charCodeAt(i+3)-97);
        s+=String.fromCharCode(c);
    }
    return s;
}
function compile(code)
{
    var c="";
    for(var i=0;i<code.length;i++){
        var b5=((code.charCodeAt(i) & (15 << 12)) >>> 12) + 97;
        var b6=((code.charCodeAt(i) & (15 << 8)) >>> 8) + 97;
        var b7=((code.charCodeAt(i) & (15 << 4)) >>> 4) + 97;
        var b8=(code.charCodeAt(i) & 15) + 97;
        c+=String.fromCharCode(b5)+String.fromCharCode(b6)+String.fromCharCode(b7)+String.fromCharCode(b8);
    }
    return c;
}

File: src/cool/util/Compile.java
package cool.util;

public class Compile {

    /*
    class Compile
    {
        public static function compile($code) {
            $code = iconv('UTF-8', 'UCS-2', $code);
            $len=strlen($code);
            $c="";
            for($n=0;$n<$len;$n++){
                $char=ord($code[$n]);
                $b1= (int)($char>>4)+97;
                $b2= (int)($char&15)+97;
                $c.=chr($b1).chr($b2);
            }
            return $c;
        }

        public static function uncompile($code){
            $len=strlen($code);
            $c="";
            for($n=0;$n<$len;$n+=2){
                $b1=(ord($code[$n])-97)<<4;
                $b2=ord($code[$n+1])-97;
                $c.=chr($b1+$b2);
            }
            $c=iconv("UCS-2","UTF-8",$c);
            return $c;

        }
    }*/


}
